### ISSUE TITLE
Github Application seems to be referencing the wrong lib curl

### ISSUE THREAD

**Issue by caperneoignis** [desktop/desktop#11516]
### Describe the bug

````
dyld: lazy symbol binding failed: Symbol not found: _curl_global_sslset
  Referenced from: /Applications/GitHub Desktop.app/Contents/Resources/app/git/libexec/git-core/git-remote-https
  Expected in: /usr/lib/libcurl.4.dylib

dyld: Symbol not found: _curl_global_sslset
  Referenced from: /Applications/GitHub Desktop.app/Contents/Resources/app/git/libexec/git-core/git-remote-https
  Expected in: /usr/lib/libcurl.4.dylib
  ````
  No matter what I do, reinstall Github, install via Brew or through the website, it won't stop saying this symbol is not found. However, after reinstalling git the error went away when calling git from the terminal. The error however, is still popping up in the Application.

### Version & OS

Open 'About GitHub Desktop' menu to see the Desktop version. Also include what operating system you are using.
Github: Version 2.6.2
OS: 10.14.6 (18G8012) "Mojave"
### Steps to reproduce the behavior

1. Go to Application
2. Click on "push to origin"
3. Error will popup.

however, if I go to terminal and do the same, there is no error. 

### Expected behavior

The application pushes to origin with no issues.

### Actual behavior

I get the error above.

### Screenshots

Add screenshots to help explain your problem, if applicable.

### Logs

````
2021-02-02T18:29:50.206Z - error: [ui] `git -c credential.helper= fetch --progress --prune origin` exited with an unexpected code: 128.
stderr:
dyld: lazy symbol binding failed: Symbol not found: _curl_global_sslset
  Referenced from: /Applications/GitHub Desktop.app/Contents/Resources/app/git/libexec/git-core/git-remote-https
  Expected in: /usr/lib/libcurl.4.dylib

dyld: Symbol not found: _curl_global_sslset
  Referenced from: /Applications/GitHub Desktop.app/Contents/Resources/app/git/libexec/git-core/git-remote-https
  Expected in: /usr/lib/libcurl.4.dylib
  ````

### Additional context

Is there any way to reference the local version of git? or specify a different location this application should be looking at?

**Comment by sergiou87** [comment]
Hi @caperneoignis! I'm sorry this is happening to you 😱 

I have a few questions:
- Is this the first time you see that error?
- Did 2.6.1 work well?
- Have you tried the [latest beta](https://central.github.com/deployments/desktop/desktop/latest/darwin?env=beta)?

We had some libcurl-related issues in 2.6.2 since it was the first time we used GitHub Actions to build our embedded git bundle, and had to do this to fix the build: https://github.com/desktop/dugite-native/pull/346

The problem was it looked for the `curl` from brew instead of the system one. In your case, it seems to be the other way around: it's looking for the system libcurl but it can't find it.

If the latest beta and 2.6.1 don't work either, I'd recommend to try this:
1. Uninstall curl from brew using `brew uninstall curl`. And then if that's not enough to make the app work, try the next option.
2. (Re)install Xcode's command line tools: `sudo xcode-select install`

Please, let me know if any of the above helped you fix the issue 🤞  and thank you for your patience! ❤️

**Comment by caperneoignis** [comment]
@sergiou87 yes, 2.6.1 worked well. I did try the latest beta and had the same result. That was the first time I saw the error. I changed over to ssl and the error went away. I actually didn't have curl installed when this first happened. I installed curl and still had the same issue. The system curl actually does exist at that location. It's like the system curl is missing something.

 is there a way to point github desktop to the brew curl?

**Comment by caperneoignis** [comment]
I can see if 2.6.1 still works, if there was another update from some where else that messed this up. because it just happened yesterday, so let me check if the previous version still works.

**Comment by dolthead** [comment]
I get the same error with 2.6.2 and 2.6.3-beta2. I've tried many nearly-related suggestions to no avail. Glad to see this issue open. My issue started about a week or more ago, maybe when I updated desktop?

Mac Book Pro 2018
macOS Mojave 10.14.6

```
>> brew config
HOMEBREW_VERSION: 2.7.7
ORIGIN: https://github.com/Homebrew/brew
HEAD: cd191a7f6c4c8e053336d0ea34a9886b12d06470
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: aac7ab3b3a3ed3aecdffab8e6427b07c1da18e90
Core tap last commit: 20 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 11.0 build 1100
Git: 2.30.0 => /usr/local/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 11.0.10, 11.0.2, 1.8.0_202
macOS: 10.14.6-x86_64
CLT: 11.3.1.0.1.1576735732
Xcode: 11.3.1
```

I'm happy to help debug further.

**Comment by sergiou87** [comment]
Thank you for your help @caperneoignis!

> The system curl actually does exist at that location.

You mean it exists at `/usr/lib/libcurl.4.dylib`? I ask because I don't have that and the app works fine:
```
$ ls /usr/lib/libcurl.4.dylib
ls: /usr/lib/libcurl.4.dylib: No such file or directory
```

The only (relevant) references I have found to libcurl.4 are in `/System/Volumes/Data/Library/Developer/CommandLineTools/`:
```
$ find /System/Volumes/Data/Library/Developer/CommandLineTools/ -name "libcurl*.tbd"
/System/Volumes/Data/Library/Developer/CommandLineTools//SDKs/MacOSX11.1.sdk/usr/lib/libcurl.tbd
/System/Volumes/Data/Library/Developer/CommandLineTools//SDKs/MacOSX11.1.sdk/usr/lib/libcurl.4.tbd
/System/Volumes/Data/Library/Developer/CommandLineTools//SDKs/MacOSX11.1.sdk/usr/lib/libcurl.3.tbd
/System/Volumes/Data/Library/Developer/CommandLineTools//SDKs/MacOSX10.15.sdk/usr/lib/libcurl.tbd
/System/Volumes/Data/Library/Developer/CommandLineTools//SDKs/MacOSX10.15.sdk/usr/lib/libcurl.4.tbd
/System/Volumes/Data/Library/Developer/CommandLineTools//SDKs/MacOSX10.15.sdk/usr/lib/libcurl.3.tbd
```

Did you reinstall the command line tools with `xcode-select --install`?

> is there a way to point github desktop to the brew curl?

AFAIK there isn't (specially since the app binaries are signed). Whatever shared lib is referenced at compile time will be loaded when you use the app.

Could you run these commands for me to see what output you get? (With the system curl, if you have curl from Homebrew you might need to uninstall it)

```
$ which curl
/usr/bin/curl
$ curl --version
curl 7.64.1 (x86_64-apple-darwin20.0) libcurl/7.64.1 (SecureTransport) LibreSSL/2.8.3 zlib/1.2.11 nghttp2/1.41.0
Release-Date: 2019-03-27
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: AsynchDNS GSS-API HTTP2 HTTPS-proxy IPv6 Kerberos Largefile libz MultiSSL NTLM NTLM_WB SPNEGO SSL UnixSockets
```
```
$ which curl-config
/usr/bin/curl-config
$ curl-config --version
libcurl 7.64.1
$ curl-config --cflags

$ curl-config --libs
-lcurl
$ curl-config --built-shared
yes
$ curl-config --prefix
/usr
```

What you see above is the output I get from those command on my macOS Big Sur.

**Comment by dolthead** [comment]
Here is mine (assuming it is helpful):
```
~ $which curl
/usr/bin/curl
~ $curl --version
curl 7.54.0 (x86_64-apple-darwin18.0) libcurl/7.54.0 LibreSSL/2.6.5 zlib/1.2.11 nghttp2/1.24.1
Protocols: dict file ftp ftps gopher http https imap imaps ldap ldaps pop3 pop3s rtsp smb smbs smtp smtps telnet tftp 
Features: AsynchDNS IPv6 Largefile GSS-API Kerberos SPNEGO NTLM NTLM_WB SSL libz HTTP2 UnixSockets HTTPS-proxy 
~ $which curl-config
/usr/bin/curl-config
~ $curl-config --version
libcurl 7.54.0
~ $curl-config --cflags

~ $curl-config --libs
-lcurl
~ $curl-config --built-shared
yes
~ $curl-config --prefix
/usr
```

I reinstalled command line tools:
```
sudo rm -rf /Library/Developer/CommandLineTools
sudo xcode-select --install
```

**Comment by caperneoignis** [comment]
@sergiou87 all command came back the same as yours except the version.
curl version: libcurl 7.54.0
curl-config version: libcurl 7.54.0

I found libcurl.4.dylib
in /usr/lib
/Library/McAfee/agent/lib/lib64/tools/
/Library/Developer/CommandLineTools/SDK/MacOSX.sdk/usr/lib/
/usr/local/Cellar/curl/7.75.0/lib/

**Comment by caperneoignis** [comment]
let me try reinstalling xcode

**Comment by sergiou87** [comment]
I'm starting to consider the problem comes from building git with a more modern Xcode or macOS SDK. I started to do some digging in desktop/dugite-native#350 and will let you know if I have something to test.

Sorry about all these problems and thanks for the help debugging it!! I really appreciate it ❤️

**Comment by caperneoignis** [comment]
yea, reinstalled xcode and still get the same error. let me see if rolling back versions help.

**Comment by sergiou87** [comment]
@caperneoignis @dolthead I just made a build which has git built with macOS Mojave's SDK. If you have a chance, please give it a try and let me know if it works: https://s3.amazonaws.com/github-desktop/releases/2.6.3-test4-d8def0e1/GitHubDesktop.zip

The build comes from the branch [releases/2.6.3-test4](https://github.com/desktop/desktop/commits/releases/2.6.3-test4), if you want to take a look, although the important bits are in desktop/dugite-native#350

Make sure to place it in your `/Applications` folder to avoid issues (but feel free to rename it to something else like `GitHub Desktop-test` instead of replacing your current version).

Thank you!

**Comment by dolthead** [comment]
@sergiou87 That build indeed fixed the issue for me. Thank you!

**Comment by sergiou87** [comment]
Thank you for testing it!! ❤️  I'm so happy it worked :_D 🎉 🎉 🎉 

~~I'll wait for confirmation from @caperneoignis before merging this fix into our main branch.~~ **UPDATE:** We decided to go ahead anyway releasing a beta, but I'd love to hear if it worked for @caperneoignis anyway 🤞 

I wonder if users of older versions of macOS had this issue and, if so, whether this new build fixes it for them too 🤔

**Comment by sergiou87** [comment]
We just released [a new beta 2.6.3-beta3](https://central.github.com/deployments/desktop/desktop/latest/darwin?env=beta) that should include this fix too 😄

**Comment by caperneoignis** [comment]
@sergiou87 crap sorry, about that got super busy. I'll try the fix here soon. been out of the office. If I get any further issues, I'll reopen an issue. Thank you for pushing an update so fast. Sorry, real life has me pulled in a bunch of directions.
